source/common/http/conn_pool_base.cc

```diff
@@ -58,22 +58,32 @@ HttpConnPoolImplBase::~HttpConnPoolImplBase() { destructAllConnections(); }
 
 ConnectionPool::Cancellable*
 HttpConnPoolImplBase::newStream(Http::ResponseDecoder& response_decoder,
-                                Http::ConnectionPool::Callbacks& callbacks) {
+                                Http::ConnectionPool::Callbacks& callbacks, bool has_early_data,
+                                bool should_use_alt_svc) {
   HttpAttachContext context({&response_decoder, &callbacks});
-  return newStreamImpl(context);
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_alt_svc") &&
+      !should_use_alt_svc && onlySupportsAltSvc()) {
```

Comment:
nit: maybe "isAlternateProtocol()" would be better than "onlySupportsAltSvc()"? what do you think?